### PR TITLE
Migrate the SwiftShader build to Vulkan.

### PR DESCRIPTION
**Pull request by prideout**
This also adds a -t flag to the easy build script for convenience.

SwiftShader on Vulkan can now be used as follows.

    git clone https://github.com/google/swiftshader.git
    cd swiftshader/build
    cmake .. &&  make -j && export SWIFTSHADER_LD_LIBRARY_PATH=`pwd`
    cd ../filament
    ./build.sh -t debug && ./out/cmake-debug/samples/gltf_viewer -a vulkan

Note that `third_party/swiftshader` does not need to exist since we already have Vulkan headers. I will remove it in a subsequent PR.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -484,7 +483,11 @@ endif()
 # "bluevk" and "samples" targets.
 # ==================================================================================================
 
-if (FILAMENT_SUPPORTS_VULKAN AND APPLE)
+if (FILAMENT_USE_SWIFTSHADER)
```
Comment (bejado) [inline CMakeLists.txt:486]: Should we verify that `FILAMENT_SUPPORTS_VULKAN` is set here? SwiftShader is only useful with Vulkan support, correct?

File: filament/backend/src/Platform.cpp [hunk 2]
```diff
@@ -35,7 +33,7 @@
         #include "vulkan/PlatformVkCocoaTouch.h"
     #endif
 #elif defined(__APPLE__)
-    #ifndef FILAMENT_USE_EXTERNAL_GLES3
+    #if !defined(FILAMENT_USE_EXTERNAL_GLES3) && !defined(FILAMENT_USE_SWIFTSHADER)
```
Comment (bejado) [inline filament/backend/src/Platform.cpp:36]: Why is this change is only necessary for APPLE platforms?

File: CMakeLists.txt (hunk 1 above)
Comment (prideout) [reply CMakeLists.txt:486]: Good idea, added an error message here.

File: filament/backend/src/Platform.cpp (hunk 2 above)
Comment (prideout) [reply filament/backend/src/Platform.cpp:36]: Excellent catch, added this for all desktop platforms.

**Pull request merged**